ibis/expr/operations.py

```diff
@@ -1967,6 +1967,11 @@ def output_type(self):
         return rules.shape_like_args(self.args, 'boolean')
 
 
+class Not(UnaryOp):
+
+    output_type = rules.shape_like_arg(0, 'boolean')
```

Comment:
Actually, this is guaranteed by only adding this method to boolean expressions in `api.py`, but I think it's more readable if we specify `input_type` here as well.